parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnWriterV1.java

```diff
@@ -109,9 +114,11 @@ private void accountForValueWritten() {
           + dataColumn.getBufferedSize();
       if (memSize > pageSizeThreshold) {
         // we will write the current page and check again the size at the predicted middle of next page
-        valueCountForNextSizeCheck = valueCount / 2;
+        if(estimateNextSizeCheck) {
```

Comment:
The behavior seems odd to me. If `estimateNextSizeCheck` is false, then this will calculate the in-memory size for every value that gets written. At a minimum, it seems like that should be done every `initialRowCountForSizeCheck` values, not every value. That initial size must be small enough to not overshoot the target, so I think this strategy is a reasonable trade-off.

I also want to make sure we definitely need this option because it looks like something that isn't particularly useful. Why is the initial row count check insufficient? Is the variance of encoded size too large?